forge/csrc/graph_lib/utils.cpp

```diff
@@ -2245,6 +2245,11 @@ bool is_consteval_capable_input_no_operand_forks(Graph *graph, InputNode *input)
     std::vector<Node *> users = graph->data_users(input);
     std::vector<Edge> user_edges = graph->user_data_edges(input);
 
+    if (input->requires_grad() && graph->training())
```

Comment:
Do we need to check for both? 

What is the case where these can differ?

Reply:
Generally, `requires_grad` is set depending on how the framework module is defined. By default, all `torch.nn.Parameter`s in torch are marked as `requires_grad` - regardless of training or inference mode.

So, if I only check for `requires_grad` then I'll disable consteval for params during inference (which is not ideal).